RecoTracker/TkDetLayers/src/TBLayer.h

```diff
@@ -25,14 +25,14 @@ class TBLayer: public BarrelDetLayer {
 
   // GeometricSearchDet interface
   
-  virtual const std::vector<const GeomDet*>& basicComponents() const final {return theBasicComps;}
+  virtual const std::vector<const GeomDet*>& basicComponents() const {return theBasicComps;}
```

Comment:
Sorry for the misunderstanding. This discussion can be related with your other comment 
"there are also theBasicComps in the base class.
Is it needed here as well?"
and the answer is yes, I am going to remove the methods from Phase2OTtiltedBarrelLayer and keeping the "final" in TBLayer.